src/header.rs

```diff
@@ -276,23 +276,24 @@ pub(crate) fn file_read_header(file: &mut File, offset: u64) -> Result<Header> {
 
     let h = Header {
         signature: sigstr.to_string(),
-        revision: u32::from_le_bytes( read_exact_buff!(rev, reader, 4) ),
-        header_size_le: u32::from_le_bytes( read_exact_buff!(hsle, reader, 4) ),
-        crc32: u32::from_le_bytes( read_exact_buff!(crc32, reader, 4) ),
-        reserved: u32::from_le_bytes( read_exact_buff!(reserv, reader, 4) ),
-        current_lba: u64::from_le_bytes( read_exact_buff!(clba, reader, 8) ),
-        backup_lba: u64::from_le_bytes( read_exact_buff!(blba, reader, 8) ),
-        first_usable: u64::from_le_bytes( read_exact_buff!(fusable, reader, 8) ),
-        last_usable: u64::from_le_bytes( read_exact_buff!(lusable, reader, 8) ),
+        revision: u32::from_ne_bytes(read_exact_buff!(rev, reader, 4)),
+        header_size_le: u32::from_ne_bytes(read_exact_buff!(hsle, reader, 4)),
+        crc32: u32::from_ne_bytes(read_exact_buff!(crc32, reader, 4)),
+        reserved: u32::from_ne_bytes(read_exact_buff!(reserv, reader, 4)),
+        current_lba: u64::from_ne_bytes(read_exact_buff!(clba, reader, 8)),
+        backup_lba: u64::from_ne_bytes(read_exact_buff!(blba, reader, 8)),
+        first_usable: u64::from_ne_bytes(read_exact_buff!(fusable, reader, 8)),
+        last_usable: u64::from_ne_bytes(read_exact_buff!(lusable, reader, 8)),
         disk_guid: parse_uuid(&mut reader)?,
-        part_start: u64::from_le_bytes( read_exact_buff!(pstart, reader, 8) ),
-        num_parts: u32::from_le_bytes( read_exact_buff!(nparts, reader, 4) ),
-        part_size: u32::from_le_bytes( read_exact_buff!(partsize, reader, 4) ),
-        crc32_parts: u32::from_le_bytes( read_exact_buff!(crc32parts, reader, 4) ),
+        part_start: u64::from_ne_bytes(read_exact_buff!(pstart, reader, 8)),
+        // Note: this will always return the total number of partition entries
+        // in the array, not how many are actually used
+        num_parts: u32::from_ne_bytes(read_exact_buff!(nparts, reader, 4)),
+        part_size: u32::from_ne_bytes(read_exact_buff!(partsize, reader, 4)),
+        crc32_parts: u32::from_ne_bytes(read_exact_buff!(crc32parts, reader, 4)),
```

Comment:
All these endianness changes seems unrelated to the topic of this PR. Why are you changing it?

Reply:
GPT spec is little endian. I don't understand why these changes.

Reply:
I can remove those -> I was initially concerned about the endianness (I thought that was why the partition thing was having issues, I can remove those if you'd like?)

Reply:
Alright, I returned those from_ne_bytes back to from_le_bytes